censys/exceptions.py

```diff
@@ -19,48 +19,266 @@ class CensysCLIException(CensysException):
 
 class CensysAPIException(CensysException):
     """
-    Base Exception for the Censys API.
+    Base Exception for Censys API's.
     """
 
+    # pylint: disable=too-many-arguments
     def __init__(
         self,
         status_code: int,
         message: str,
         body: Optional[str] = None,
         const: Optional[str] = None,
+        error_code: Optional[int] = None,
+        details: Optional[str] = None,
     ):
         self.status_code = status_code
         self.message = message
         self.body = body
         self.const = const
+        self.error_code = error_code
+        self.details = details
         super().__init__(self.message)
 
+
+class CensysSearchException(CensysAPIException):
+    """
+    Base Exception for the Censys search API.
+    """
+
     def __repr__(self):
         return "%i (%s): %s" % (self.status_code, self.const, self.message or self.body)
 
     __str__ = __repr__
 
 
-class CensysRateLimitExceededException(CensysAPIException):
+class CensysAsmException(CensysAPIException):
```

Comment:
This should be in `censys.asm.exceptions` it certainly can also be imported here, but I generally want to avoid `*` imports if possible.